lib/rucio/core/heartbeat.py

```diff
@@ -70,6 +83,9 @@ def _sanity_check(executable, hostname, hash_executable=None, session=None):
             if not pid_exists(pid):
                 session.query(Heartbeats).filter_by(hostname=hostname, pid=pid).delete()
 
+    if expiration_delay:
+        cardiac_arrest(older_than=expiration_delay, session=session)
```

Comment:
I wonder if we should put this into a try/except. Reason being that the `sanity_check` is often called in a daemon outside of the while loop which would recover from errors. And there might be some row-contention/deadlock errors if this is called at the same time.

Reply:
_sanity_check is wrapped by sanity_check() which does exactly that. Do we need to catch any exceptions other than "databaseError" ?

Reply:
Right, I have misread this, ignore what I said :-)